packages/cozy-stack-client/src/Collection.js

```diff
@@ -0,0 +1,51 @@
+import { uri } from './utils'
+
+/**
+ * Handler for error response which return a empty value for "not found" error
+ * @param  {Error} error
+ * @param  {Array|Object}  [data=[]] Data to return in case of "not found" error
+ * @return {Object}        JsonAPI response with empty data in case of "not
+ * found" error.
+ */
+export const dontThrowNotFoundError = (error, data = []) => {
+  if (error.message.match(/not_found/)) {
+    return Array.isArray(data)
+      ? { data, meta: { count: 0 }, skip: 0, next: false }
+      : {
+          data
+        }
+  }
+  throw error
+}
+
+/**
+ * Utility class to abstract an regroup identical methods and logics for
+ * specific collections.
+ */
+export class Collection {
+  /**
+   * Utility method aimed to return only one document.
+   * @param  {Object}  stackClient
+   * @param  {String}  id              Document identifier
+   * @param  {String}  endpoint        Stack endpoint
+   * @param  {Func}    normalize       Callback to normalize response data
+   * @param  {String}  method          HTTP method
+   * @return {Promise}                 Promise of JsonAPI response
```

Comment:
As the function is async, it is already infered that you return a promise. @return should declare what you return inside that promise